examples/tensorflow/text-classification/run_text_classification.py

```diff
@@ -0,0 +1,532 @@
+        # Now we've established our label2id, let's overwrite the model config with it.
+        model.config.label2id = label_to_id
+        if model.config.label2id is not None:
+            model.config.id2label = {id: label for label, id in label_to_id.items()}
+        else:
+            model.config.id2label = None
+    else:
+        label_to_id = model.config.label2id  # Just load the data from the model
+
+    if "validation" in datasets and model.config.label2id is not None:
+        validation_label_list = datasets["validation"].unique("label")
+        for val_label in validation_label_list:
+            assert val_label in label_to_id, f"Label {val_label} is in the validation set but not the training set!"
+
+    def preprocess_function(examples):
+        # Tokenize the texts
+        args = (
+            (examples[sentence1_key],) if sentence2_key is None else (examples[sentence1_key], examples[sentence2_key])
+        )
+        result = tokenizer(*args, padding=padding, max_length=max_seq_length, truncation=True)
+
+        # Map labels to IDs
+        if model.config.label2id is not None and "label" in examples:
+            result["label"] = [(model.config.label2id[l] if l != -1 else -1) for l in examples["label"]]
+        return result
+
+    datasets = datasets.map(preprocess_function, batched=True, load_from_cache_file=not data_args.overwrite_cache)
+
+    if "train" in datasets:
+        train_dataset = datasets["train"]
+        if data_args.max_train_samples is not None:
+            train_dataset = train_dataset.select(range(data_args.max_train_samples))
+        # Log a few random samples from the training set so we can see that it's working as expected:
+        for index in random.sample(range(len(train_dataset)), 3):
+            logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
+
+    if "validation" in datasets:
+        eval_dataset = datasets["validation"]
+        if data_args.max_val_samples is not None:
+            eval_dataset = eval_dataset.select(range(data_args.max_val_samples))
+
+    if "test" in datasets:
+        test_dataset = datasets["test"]
+        if data_args.max_test_samples is not None:
+            test_dataset = test_dataset.select(range(data_args.max_test_samples))
+
+    # endregion
+
+    # region Training
+    if "train" in datasets:
+        training_dataset = DataSequence(
+            train_dataset, non_label_column_names, batch_size=training_args.per_device_train_batch_size, labels=True
+        )
+        if "validation" in datasets:
+            eval_dataset = DataSequence(
+                eval_dataset, non_label_column_names, batch_size=training_args.per_device_eval_batch_size, labels=True
+            )
+        else:
+            eval_dataset = None
+
+        callbacks = [SavePretrainedCallback(output_dir=training_args.output_dir)]
+        model.fit(training_dataset, validation_data=eval_dataset, epochs=10, callbacks=callbacks)
```

Comment:
Is it on purpose that `epochs=10` is hardcoded? shouldn't this be `epochs=training_args. num_train_epochs`?

Reply:
Aagh, you're totally right. Fixing it!